pkg/applyset/applyset.go

```diff
@@ -0,0 +1,546 @@
+// Copyright 2025 The Kube Resource Orchestrator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Applylib is inspired from:
+//  * kubectl pkg/cmd/apply/applyset.go
+//  * kubebuilder-declarative-pattern/applylib
+//  * Creating a simpler, self-contained version of the library that is purpose built for controllers.
+//  * KEP describing applyset:
+//     https://git.k8s.io/enhancements/keps/sig-cli/3659-kubectl-apply-prune#design-details-applyset-specification
+
+package applyset
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/go-logr/logr"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/dynamic"
+)
+
+type ToolingID struct {
+	Name    string
+	Version string
+}
+
+func (t ToolingID) String() string {
+	return fmt.Sprintf("%s/%s", t.Name, t.Version)
+}
+
+type Set interface {
+	Add(ctx context.Context, obj ApplyableObject) error
```

Comment:
This interface needs to be heavily documented.

Reply:
added initial docs. PTAL.